### PR TITLE
Emit Cloudwatch metric to show CI Status

### PR DESCRIPTION
**Pull request by rapphil**
Signed-off-by: Raphael Silva <rapphil@gmail.com>

**Description:** Emit CI status as CloudWatch metric.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -726,6 +726,25 @@ jobs:
           cd testing-framework/terraform
           make checkCacheHits
 
+  update-ci-status:
+    runs-on: ubuntu-latest
+    needs: [validate-all-tests-pass]
```
Comment (bryan-aguilar) [inline .github/workflows/CI.yml:731]: Should we move this to have a dependency on `release-candidate`? There are still steps that could fail during that job.

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -726,6 +726,25 @@ jobs:
...
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.COLLECTOR_ASSUMABLE_ROLE_ARN }}
+          aws-region: us-west-2
+
+      - name: Publish CI status
+        run: |
+          if [ '${{needs.validate-all-tests-pass.result}}' = 'success' ]; then
+            aws cloudwatch put-metric-data --namespace 'ADOT/CI' --metric-name Success --dimensions repository=aws-otel-collector --value 1.0
```
Comment (bryan-aguilar) [inline .github/workflows/CI.yml:743]: Can we add the branch name as a dimension also? Could help with filtering out release or dev branches.

Comment (bryan-aguilar) [reply .github/workflows/CI.yml:743]: `${{ github.ref_name }}`

Comment (rapphil) [reply .github/workflows/CI.yml]: Good idea. done

File: .github/workflows/CI.yml (hunk 1 above)
Comment (rapphil) [reply .github/workflows/CI.yml]: Good point. moved to depend on the release candidate job.

**Review by bryan-aguilar — APPROVED** [review]
Looks good, can we hold off on merging until the current releases are through the pipeline?

**Pull request merged**